google/cloud/forseti/notifier/notifiers/cscc_notifier.py

```diff
@@ -187,6 +188,56 @@ def _transform_for_api(self, violations, source_id=None):
             findings.append(finding)
         return findings
 
+    @classmethod
+    def find_stale_findings(cls, new_findings, findings_in_cscc):
+        """Finds the findings that does not correspond to the latest scanner run
+        and updates it's state to inactive.
+
+        Args:
+            new_findings (list): Latest violations that are transformed to
+            findings.
+            findings_in_cscc (list): Findings pulled from CSCC UI that
+            corresponds to the previous scanner run.
+
+        Returns:
+            list: Findings whose state has been marked as 'INACTIVE'.
+        """
+
+        inactive_findings = []
+        new_findings_map = {}
+
+        for finding_list in new_findings:
+            finding_id = finding_list[0]
+            finding = finding_list[1]
+            add_to_dict = {finding_id: finding}
+            new_findings_map.update(add_to_dict)
+
+        for finding_list in findings_in_cscc:
+            finding_id = finding_list[0]
+            to_be_updated_finding = finding_list[1]
+
+            # Removing keys that are not part of the Finding data structure.
+            to_be_updated_finding.pop('securityMarks')
+            to_be_updated_finding.pop('createTime')
+
+            # Renaming keys to match the keys in the Finding data structure.
+            to_be_updated_finding['source_properties'] = (
+                to_be_updated_finding.pop('sourceProperties'))
```

Comment:
I don't understand why you have to pop these attributes.  We are using update field mask?  So, any other fields should not be relevant?

Reply:
Removed the code to pop keys.